apim-operator/pkg/apis/wso2/v1alpha1/security_types.go

```diff
@@ -0,0 +1,54 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SecuritySpec defines the desired state of Security
+// +k8s:openapi-gen=true
+type SecuritySpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Type        string `json:"type"`
+	Certificate string `json:"certificate"`
+	Alias       string `json:"alias"`
+	Endpoint    string `json:"endpoint"`
+	Credentials string `json:"credentials"`
+}
+
+// SecurityStatus defines the observed state of Security
+// +k8s:openapi-gen=true
+type SecurityStatus struct {
```

Comment:
having a struct without any fields is done intentionally?

Reply:
No this is autogenerated